sdl_viewer/src/lib.rs

```diff
@@ -196,31 +196,10 @@ impl PointCloudRenderer {
             }
         }
 
-        // Bisect the actual level to choose, we want to be as close as possible to the max
-        // nodes to use.
-        let mut max_level_to_display =
-            if moving { self.max_level_moving } else { 256 };
-        let mut min_level_to_display = 0;
-        let mut filtered_visible_nodes: Vec<_>;
-        let mut max_level_seen = 0;
-        while (max_level_to_display - min_level_to_display) > 1 {
-            let current = (max_level_to_display + min_level_to_display) / 2;
-            filtered_visible_nodes = self.visible_nodes
-                .iter()
-                .inspect(|id| max_level_seen = max_level_seen.max(id.level()))
-                .filter(|id| id.level() <= current)
-                .collect();
-            if filtered_visible_nodes.len() > self.max_nodes_in_memory {
-                max_level_to_display = current;
-            } else {
-                min_level_to_display = current;
-            }
-        }
-        filtered_visible_nodes = self.visible_nodes
-            .iter()
-            .filter(|id| id.level() <= min_level_to_display)
-            .collect();
-        assert!(filtered_visible_nodes.len() < self.max_nodes_in_memory);
+        let max_nodes_to_display =
```

Comment:
Add a comment that we use a heuristic to find the correct number of nodes to keep the framerate decent.